### ISSUE TITLE
Video with low source bitrate transcodes into gray rectangle

### ISSUE THREAD

**Issue by scanlime** [Chocobozzz/PeerTube#4516]
**Describe the current behavior**

I got this report from a diode zone user, they uploaded a video file and discovered that it plays back as a gray rectangle:

https://diode.zone/w/tSqf6Tovvk1zTKZvVYxZZd

![image](https://user-images.githubusercontent.com/923812/140173392-a90d538a-0973-48d1-83e6-0d6eec1d3f15.png)

The original video file is quite small, attached here:

https://user-images.githubusercontent.com/923812/140173627-f8040610-6188-49c2-b6b7-a81c9a98521e.mp4

```
micah@tuber:~/tmp$ sha256sum eZh.mp4 
a2de8598dfe037addf16357f674248bbd1fc70f3917050cdf420cfc6ff4c714b  eZh.mp4
micah@tuber:~/tmp$ ffprobe eZh.mp4 
ffprobe version 4.2.4-1ubuntu0.1 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-10ubuntu2)
  configuration: --prefix=/usr --extra-version=1ubuntu0.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-avisynth --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-nvenc --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'eZh.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    encoder         : Lavf58.76.100
  Duration: 00:00:24.00, start: 0.000000, bitrate: 7 kb/s
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 1920x1080, 6 kb/s, 1 fps, 1 tbr, 16384 tbn, 2 tbc (default)
    Metadata:
      handler_name    : VideoHandler
```

**Steps to reproduce:**

1. Upload this video
2. Let it transcode
3. Gray rectangle

Specifically, the output video also has a very low bitrate (6 kbps) and I suspect the problem is that we are deriving an output bitrate from the input, but it's too low to do an effective job at transcoding with these settings.

**Describe the expected behavior**

Expect there to be a reasonable lower limit on the output bitrate for the transcoded video.

**Additional information**

* PeerTube instance:
  * URL: diode.zone
  * version: 3.4.1
  * NodeJS version:
  * Ffmpeg version: 4.2.4-1ubuntu0.1

* Browser name, version and platforms on which you could reproduce the bug:
* Link to browser console log if relevant:
* Link to server log if relevant (`journalctl` or `/var/www/peertube/storage/logs/`):

**Comment by andinus** [comment]
I uploaded the original video. It was generated with:

```
ffmpeg -loglevel "$log-level" -r "$frame-rate" -i "$output/\%08d.png"
                -vcodec libx264 -crf 28 -pix_fmt yuv420p "$output/solution.mp4"
```
- `$frame-rate` was set to 1.

- https://github.com/andinus/fornax/blob/203ba71bc2b5bded2bc5c4a9e1d889b483e7f30f/lib/Fornax/CLI.rakumod#L135

**Comment by scanlime** [comment]
1fps seems like a great way to trigger this bug, since peertube will require keyframes every other frame and 6kbps won't go very far. Maybe a minimum frame rate would also be useful (and have ffmpeg insert duplicates as necessary) since some formats act strangely at these extremes. A higher minimum bitrate seems like the simplest fix for this issue though.

**Comment by Chocobozzz** [comment]
I added a minimum bitrate in https://github.com/Chocobozzz/PeerTube/commit/67eeec8b955339120ff5d3c8286fdf0715e6270c

It seems to fix the issue on my side (ffmpeg 4.4.1).
I'm not really sure about the `minLimitBitPerPixel` values, so don't hesitate to tell me if you would have used other ones.
